astropy/stats/circstats.py

```diff
@@ -0,0 +1,448 @@
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
+    .. [3] M. Chirstman., C. Miller. "Testing a Sample of Directions for
+       Uniformity." Lecture Notes, STA 6934/5805. University of Florida, 2007.
+    .. [4] D. Wilkie. "Rayleigh Test for Randomness of Circular Data". Applied
+       Statistics. 1983.
+       <http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.211.4762>
+    """
+    n = np.size(data, axis=axis)
+    Rbar = _length(data, w, 1, 0.0, axis)
+    z = n*Rbar*Rbar
+
+    # see [3] and [4] for the formulae below
+    tmp = 1.0
+    if(n < 50):
+        tmp = 1.0 + (2.0*z - z*z)/(4.0*n) - (24.0*z - 132.0*z**2.0 +
+                                             76.0*z**3.0 - 9.0*z**4.0)/(288.0 *
+                                                                        n * n)
+
+    p_value = np.exp(-z)*tmp
+    return p_value
+
+
+def vtest(data, w=None, mu=0.0, axis=None):
+    """ Performs the Rayleigh test of uniformity where the alternative
+    hypothesis H1 is assumed to have a known mean angle ``mu``.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array of circular (directional) data.
+    w : numpy.ndarray, optional
+        In case of grouped data, the i-th element of ``w`` represents a
+        weighting factor for each group such that sum(w,axis) equals the number
+        of observations. See [1], remark 1.4, page 22, for detailed
+        explanation.
+    mu : float, optional
+        Mean angle. Assumed to be known.
+    axis : int, optional
+        Axis along which the V test will be performed.
+
+    Returns
+    -------
+    p-value : float
+        p-value.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.stats import vtest
+    >>> from astropy import units as u
+    >>> data = np.array([130, 90, 0, 145])*u.deg
+    >>> vtest(data) # doctest: +FLOAT_CMP
+    <Quantity 0.6223678199713766>
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
+    .. [3] M. Chirstman., C. Miller. "Testing a Sample of Directions for
+       Uniformity." Lecture Notes, STA 6934/5805. University of Florida, 2007.
+    """
+    from scipy.stats import norm
+
+    if w is None:
+        w = np.ones(data.shape)
+    elif (w.shape != data.shape):
+        raise ValueError('shape of w %s is not equals shape of data %s.'
+                         % (w.shape, data.shape,))
+
+    n = np.size(data, axis=axis)
+
+    R0bar = np.sum(w*np.cos(data-mu), axis)/np.sum(w, axis)
```

Comment:
In analogy with above, this could be
```
R0bar = np.average(np.cos(data-mu), weights=w, axis=axis)
```
in which case the tests on `w` can be removed.